### ISSUE TITLE
Wiki abbreviation redirects are incorrectly parsed as locale codes

### ISSUE THREAD

**Issue by TicClick** [ppy/osu-web#7035]
One case found by pure accident is that https://osu.ppy.sh/wiki/AR will always redirect you to the **ar**abic version of the article, https://osu.ppy.sh/wiki/ar/Beatmapping/Approach_rate.

**Comment by nanaya** [comment]
What's the desired result here?

1. Should `/wiki/AR` redirect to `/wiki/en/Beatmapping/Approach_rate` (with `en` as current locale)?
    - and then there will be no default redirect for `/wiki/en` or other locales (currently redirects to `/wiki/en/Main_page`)
2. or have `/wiki/AR` to always redirect to `/wiki/ar/Main_Page`?

**Comment by notbakaneko** [comment]
I'm thinking a locale without a page shouldn't redirect and `wiki/AR` should redirect to the article of whatever the set language is.

**Comment by TicClick** [comment]
I'd say 1), as long as you can still refer to a specific locale, for example, the way it works now (`/wiki/ru/Accuracy` or `/wiki/en/Main Page`). I used this a couple of times when I needed to link specific translations in-game.

I also expect anchors to work:

- `/wiki/AR#whatever` -> `/wiki/<current locale>/Beatmapping/Approach_rate#whatever`
- `/wiki/SV` -> `/wiki/<current locale>/Glossary#slider-velocity` (can't remember the exact heading)

as well as references to categorized articles (so far there are none):

- `/wiki/AR/History` ->
  - `/wiki/<current locale>/Beatmapping/Approach_rate/History` if the article exists
  - `/wiki/<ar as locale>/History` if it doesn't

in other words, if it's two letters alone or with a hashtag, it's a redirect. if it's followed by a slash, it's:

- `locale + path to an article` if there's no redirect looking like `locale`
- `path to an article in the current locale` if there is such redirect.

**Comment by nanaya** [comment]
> `/wiki/AR/History` ->
> 
>     * `/wiki/<current locale>/Beatmapping/Approach_rate/History` if the article exists
> 
>     * `/wiki/<ar as locale>/History` if it doesn't

Doesn't that mean if `/wiki/ar/History` actually exists, it's not accessible?

Would a third option, redirect prefix be better? `/wiki/r/ar` etc

Seems like some kind of hell mixing redirect and locale :raised_eyebrow:

**Comment by TicClick** [comment]
> Doesn't that mean if `/wiki/ar/History` actually exists, it's not accessible?
> 
> Would a third option, redirect prefix be better? `/wiki/r/ar` etc
> 
> Seems like some kind of hell mixing redirect and locale 🤨

yeah, it is a drawback, which is very unlikely to be observed: you need a top-level article to exist and use abbreviation as a title, which is [forbidden by the styling criteria](https://github.com/ppy/osu-wiki/blob/master/wiki/Article_styling_criteria/Formatting/en.md#article-naming) for regular articles and allowed only for tournament-related pages, which already reside in `/wiki/Tournaments`.

a special path component for redirects allows for clear distinction between a redirect and a regular article, but is less convenient.

at least these are my thoughts and expectations from the viewpoint of a wiki user/contributor
